File: ProjetScolaire -back/src/main/java/ProjetScolaire/entity/Adresse.java
package ProjetScolaire.entity;

import javax.persistence.Embeddable;

import com.fasterxml.jackson.annotation.JsonView;

@Embeddable
public class Adresse {
	@JsonView(Vue.Common.class)
	protected int numero;
	@JsonView(Vue.Common.class)
	protected String voie;
	
	@JsonView(Vue.Common.class)
	protected int cp;
	
	@JsonView(Vue.Common.class)
	protected String ville;
	
	public Adresse() {
		super();
	}

	public Adresse(int numero, String voie, int cp, String ville) {
		super();
		this.numero = numero;
		this.voie = voie;
		this.cp = cp;
		this.ville = ville;
	}


	public int getNumero() {
		return numero;
	}


	public void setNumero(int numero) {
		this.numero = numero;
	}


	public String getVoie() {
		return voie;
	}


	public void setVoie(String voie) {
		this.voie = voie;
	}


	public String getVille() {
		return ville;
	}


	public void setVille(String ville) {
		this.ville = ville;
	}


	public int getCp() {
		return cp;
	}

	public void setCp(int cp) {
		this.cp = cp;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + numero;
		result = prime * result + ((ville == null) ? 0 : ville.hashCode());
		result = prime * result + ((voie == null) ? 0 : voie.hashCode());
		return result;
	}


	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Adresse other = (Adresse) obj;
		if (numero != other.numero)
			return false;
		if (ville == null) {
			if (other.ville != null)
				return false;
		} else if (!ville.equals(other.ville))
			return false;
		if (voie == null) {
			if (other.voie != null)
				return false;
		} else if (!voie.equals(other.voie))
			return false;
		return true;
	}
	
	

}

File: projet-scolaire-front/src/app/model/adresse.ts
export class Adresse {
  private _numero: number;
  private _voie: string;
  private _ville: string;
  private _cp: number;

  constructor(
    numero?: number,
    voie: string = '',
    ville: string = '',
    cp?: number
  ) {
    this._numero = numero;
    this._voie = voie;
    this._ville = ville;
    this._cp = cp;
  }

  /**
   * Getter numero
   * @return {number}
   */
  public get numero(): number {
    return this._numero;
  }

  /**
   * Getter voie
   * @return {string}
   */
  public get voie(): string {
    return this._voie;
  }

  /**
   * Getter ville
   * @return {string}
   */
  public get ville(): string {
    return this._ville;
  }

  /**
   * Setter numero
   * @param {number} value
   */
  public set numero(value: number) {
    this._numero = value;
  }

  /**
   * Setter voie
   * @param {string} value
   */
  public set voie(value: string) {
    this._voie = value;
  }

  /**
   * Setter ville
   * @param {string} value
   */
  public set ville(value: string) {
    this._ville = value;
  }

  /**
   * Getter cp
   * @return {number}
   */
  public get cp(): number {
    return this._cp;
  }

  /**
   * Setter cp
   * @param {number} value
   */
  public set cp(value: number) {
    this._cp = value;
  }
}
